client/src/components/atoms/ImagePicker.tsx

```diff
@@ -1,24 +1,167 @@
-import React, { FunctionComponent } from "react";
+import React, { FunctionComponent, useState } from "react";
+import Cropper from "react-easy-crop";
+import Dropzone from "react-dropzone";
 import ImageList from "./ImageList";
+import Slider from "@material-ui/core/Slider";
+
+import WarningBox from "./WarningBox";
 
 interface Props {
     onImageChange(url: string): void;
+    onUploadImg(data: string): void;
     images: string[];
     selected: string;
     isErroneous: boolean;
+    uploadedImg: string;
 }
 
+interface Dimensions {
+    width: number;
+    height: number;
+}
 const ImagePicker: FunctionComponent<Props> = (props: Props) => {
-    const { selected, images, onImageChange, isErroneous } = props;
+    const [crop, setCrop] = useState({ x: 0, y: 0 });
+    const [zoom, setZoom] = useState(1);
+    const [showWarning, setShowWarning] = useState(false);
+    const [warningText, setWarningText] = useState("");
+    const { selected, images, onImageChange, isErroneous, onUploadImg, uploadedImg } = props;
+
+    const getImageDimensions = async (file: string): Promise<Dimensions> => {
+        return new Promise((resolved) => {
+            const i = new Image();
+            i.onload = () => {
+                resolved({ width: i.width, height: i.height });
+            };
+            i.src = file;
+        });
+    };
+    const handleOnDrop = (acceptedFiles: File[]) => {
+        const file = acceptedFiles[0];
+        let validFile = true;
+        if (file.size > 6000000) {
+            triggerWarning("Images must be less than 5MB");
+            validFile = false;
+        }
+        if (file.type != "image/png" && file.type != "image/jpeg") {
+            triggerWarning("Images must be in either JPEG or PNG format");
+            validFile = false;
+        }
+        const fileReader = new FileReader();
+        fileReader.addEventListener(
+            "load",
+            async () => {
+                const res = fileReader.result;
+                console.log(res);
+                let imgStr;
+                if (res === null) imgStr = "";
+                else if (typeof res === "string") {
+                    imgStr = res;
+                } else {
+                    imgStr = res.toString();
+                }
+                const dimensions = await getImageDimensions(imgStr);
+                console.log(dimensions);
+                if (dimensions.width < 600 || dimensions.height < 430) {
+                    triggerWarning("Images must be at least 600 x 430 pixels");
+                    validFile = false;
+                }
+                if (validFile) {
+                    setShowWarning(false);
```

Comment:
Might not need here `setShowWarning` since `showWarning` is `false` by default

Reply:
`setShowWarning(false)` is here so if the user uploads a valid image the warning will disappear right away.